File: C++/A/899A-splittinginteams.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int no_teams;
    vector<int> teams;

public:
    void input()
    {
        cin >> no_teams;
        for (int i = 0; i < no_teams; i++)
        {
            int a;
            cin >> a;
            teams.push_back(a);
        }
    }
    int answer()
    {
        int cnt1 = count(teams.begin(), teams.end(), 1), cnt2 = count(teams.begin(), teams.end(), 2);
        return ((cnt1 > cnt2) ? cnt2 + (cnt1 - cnt2) / 3 : cnt1);
    }
};

int main()
{
    Solver solution;
    solution.input();
    cout << solution.answer();
}

File: Python/A/899A-splittinginteams.py
class Solver:
    def __init__(self):
        self.no_teams = 0
        self.teams = []

    def input(self):
        self.no_teams = int(input())
        self.teams = list(map(int, input().split()))

    def answer(self):
        cnt1, cnt2 = self.teams.count(1), self.teams.count(2)
        return cnt2 + (cnt1-cnt2)//3 if cnt1 > cnt2 else cnt1


solution = Solver()
solution.input()
print(solution.answer())
